jsprit-analysis/src/main/java/com/graphhopper/jsprit/analysis/toolbox/AlgorithmEventsRecorder.java

```diff
@@ -141,13 +146,16 @@ private void addRoutes(Collection<VehicleRoute> routes) {
 
     private String getNodeId(TourActivity act) {
         String nodeId = null;
-        if (act instanceof TourActivity.JobActivity) {
-            Job job = ((TourActivity.JobActivity) act).getJob();
+        if (act instanceof JobActivity) {
+            Job job = ((JobActivity) act).getJob();
             if (job instanceof Service) {
                 nodeId = job.getId();
             } else if (job instanceof Shipment) {
-                if (act.getName().equals("pickupShipment")) nodeId = getFromNodeId((Shipment) job);
-                else nodeId = getToNodeId((Shipment) job);
+                if (act.getName().equals("pickupShipment")) {
+                    nodeId = getFromNodeId((Shipment) job);
+                } else {
+                    nodeId = getToNodeId((Shipment) job);
+                }
```

Comment:
You are right. To tell you the truth, I was lazy enough to neglate the configuration of it for this project. I've used the one we use. I'll fix it tommorrow.